types/hapi-auth-cookie/index.d.ts

```diff
@@ -0,0 +1,124 @@
+// Type definitions for hapi-auth-cookie 9.1
+// Project: https://github.com/hapijs/hapi-auth-cookie
+// Definitions by: Silas Rech <https://github.com/lenovouser>
+//                 Simon Schick <https://github.com/SimonSchick>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+import { Request, ResponseObject, Plugin, ResponseToolkit } from 'hapi';
+
+declare module 'hapi' {
+    interface ServerAuth {
+        strategy(name: string, scheme: 'cookie', options?: hapiAuthCookie.Options): void;
+    }
+
+    interface Request {
+        cookieAuth: {
+            set: (session: any, value?: any) => void;
```

Comment:
This can be:

```ts
(session: any) => void;

// or

(key: 'string', value: any) => void;
```

![](https://i.lenovouser.me/cthcn2306qh4NU7k.jpg)

Any idea to type this better?

Reply:
session would be `object`, not `any`, and key would be `string` not `'string'`.
This should also be written as `set(session: object): void` imo

Reply:
@SimonSchick I did now `set(session: object | string, value?: object | string): void` now - is that okay?